test/common/buffer/watermark_buffer_test.cc

```diff
@@ -121,11 +121,8 @@ TEST_F(WatermarkBufferTest, PrependBuffer) {
 TEST_F(WatermarkBufferTest, Commit) {
   buffer_.add(TEN_BYTES, 10);
   EXPECT_EQ(0, times_high_watermark_called_);
-  RawSlice out;
-  buffer_.reserve(10, &out, 1);
-  memcpy(out.mem_, &TEN_BYTES[0], 10);
-  out.len_ = 10;
-  buffer_.commit(&out, 1);
+  Buffer::Reservation reservation = buffer_.reserveApproximately(10);
+  reservation.commit(10);
```

Comment:
Someone looking at this code may think that it is acceptable to write:
Buffer::Reservation reservation = buffer_.reserveApproximately(size);
reservation.commit(size);

I'm not sure how to avoid this hazard.

Reply:
Fixed by the removal of the size argument.